src/util.cpp

```diff
@@ -92,12 +92,28 @@ namespace rhost {
             return nc;
         }
 
+        std::wstring to_unicode(const char* buf, size_t len) {
+            // Convert 8-bit characters to Unicode via Windows CP. This guarantees
+            // if locale for non-Unicode programs is set correctly, user can type in
+            // their language. This does NOT guarantee that all languages can be used
+            // since R is not Unicode app. If host app is Unicode, it must perform
+            // checks if text being passed here is convertible to Unicode.
+            std::wstring ws;
+            size_t cch = strlen(buf);
+            if (cch > 0) {
+                ws.resize(cch);
+                RString2Unicode(&ws[0], (char*)buf, len);
```

Comment:
Also, there's a function called `strsxp_to_utf8` in json.cpp, that I think is more in spirit of what you're trying to do... we just need an equivalent that returns `wstring` rather than `string`. They could both live in util.h.